eras/conway/impl/src/Cardano/Ledger/Conway/Governance.hs

```diff
@@ -272,18 +282,42 @@ instance EraPParams era => NFData (EnactState era)
 
 instance EraPParams era => NoThunks (EnactState era)
 
+data GovActionEntry era = GovActionEntry
+  { gaeId :: !(GovActionId (EraCrypto era))
+  , gaeState :: !(GovActionState era)
+  }
+  deriving (Generic, Eq, Show)
+
+instance EraPParams era => DecCBOR (GovActionEntry era) where
+  decCBOR =
+    decode $
+      RecD GovActionEntry
+        <! From
+        <! From
+
+instance EraPParams era => EncCBOR (GovActionEntry era) where
+  encCBOR GovActionEntry {..} =
+    encode $
+      Rec GovActionEntry
+        !> To gaeId
+        !> To gaeState
+
+instance EraPParams era => NFData (GovActionEntry era)
+
+instance EraPParams era => NoThunks (GovActionEntry era)
+
+instance EraPParams era => ToJSON (GovActionEntry era)
+
 data RatifyState era = RatifyState
   { rsEnactState :: !(EnactState era)
-  , rsFuture ::
-      !( StrictSeq
-          (GovActionId (EraCrypto era), GovActionState era)
-       )
-  , rsRemoved ::
-      !( StrictSeq
-          (GovActionId (EraCrypto era), GovActionState era)
-       )
+  , rsFuture :: !(StrictSeq (GovActionEntry era))
+  , rsRemoved :: !(StrictSeq (GovActionEntry era))
```

Comment:
While looking through the PR I realized, why don't we just add `GovActionId` to the `GovActionState`? It will simplify things. The normal argument of it being duplicated in memory, since it is also used as a key in the GovActionsState Map does't really apply, since values will be shared.

@Soupstraw What do you think?
```suggestion
  , rsFuture :: !(StrictSeq (GovActionState era))
  , rsRemoved :: !(StrictSeq (GovActionState era))
```